ts/screens/authentication/CieCardReaderScreen.tsx

```diff
@@ -0,0 +1,207 @@
+import { Millisecond } from "italia-ts-commons/lib/units";
+import { Button, Content, H2, Text, View } from "native-base";
+import * as React from "react";
+import { Image, StyleSheet } from "react-native";
+import ProgressCircle from "react-native-progress-circle";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import AnimatedRing from "../../components/animations/AnimatedRing";
+import ScreenHeader from "../../components/ScreenHeader";
+import BaseScreenComponent from "../../components/screens/BaseScreenComponent";
+import I18n from "../../i18n";
+import customVariables from "../../theme/variables";
+
+interface OwnProps {
+  navigation: NavigationScreenProp<NavigationState>;
+}
+
+type Props = OwnProps;
+// Image dimension
+const imgDimension = 180;
+const boxDimension = 245;
+/** Every stepTimeProgress update circular progress, max value progress is 100, example
+ *  50: the animation ending in 16 seconds
+ *  60: the animation ending in 19 seconds
+ *  ecc ecc
+ */
+const stepTimeProgress = 50;
+
+const styles = StyleSheet.create({
+  messageHeader: {
+    minHeight: 90,
+    paddingRight: customVariables.contentPadding,
+    paddingLeft: customVariables.contentPadding,
+    paddingTop: customVariables.contentPadding,
+    fontSize: customVariables.fontSizeBase
+  },
+  messageFooter: {
+    paddingRight: customVariables.contentPadding,
+    paddingLeft: customVariables.contentPadding,
+    paddingBottom: customVariables.contentPadding,
+    fontSize: customVariables.fontSizeBase
+  },
+  titleHeader: {
+    marginTop: 35
+  },
+  imgContainer: {
+    justifyContent: "center",
+    alignItems: "center",
+    height: boxDimension
+  },
+  img: {
+    overflow: "hidden",
+    backgroundColor: customVariables.colorWhite,
+    height: imgDimension - 3,
+    width: imgDimension - 3,
+    borderRadius: imgDimension / 2
+  },
+  rings: {
+    height: boxDimension,
+    width: boxDimension,
+    position: "absolute",
+    overflow: "hidden",
+    justifyContent: "center",
+    alignItems: "center"
+  }
+});
+
+type State = {
+  progressBarValue: number;
+  // This boolean change progress color, if true the color turns red
+  isReadInterrupted: boolean;
+};
+
+/**
+ *  This screen shown while reading the card
+ */
+class CieCardReaderScreen extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      progressBarValue: 0,
+      isReadInterrupted: false
+    };
+    // Start an interval to increment progress bar
+    const interval = setInterval(() => {
```

Comment:
### avoid animate value manually
My suggestion is to avoid to make the animation "manually"
I proposed an alternative solution here f9d66db5368805b9d53ec59da439ba33c9a96564

define the animation in the constructor and start and stop it wherever you want

### state refactor
To handle all stuffs needed when a CIE card is reading we need to have this reading state:

- **waiting_card** (we are ready for read ->radar effect)
- **reading** (we are reading the card -> progress animation)
- **error** (the reading is interrupted -> progress animation stops and the progress circle becomes red)
- **completed** (the reading has been completed)